### PR TITLE
Move edm::ErrorSummaryEntry to DataFormats/Common

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Keep edm::ErrorSummaryEntry for storage into the Event and create a new edm::messagelogger::ErrorSummaryEntry as part of MessageLogger. This will hopefully make it easier to migrate to ROOT's module system as we can avoid generating a dictionary for FWCore/MessageLogger.

#### PR validation:

The code compiles.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@davidlange6 FYI

**Comment by Dr15Jones** [comment]
please test

File: DataFormats/Common/interface/ELseverityLevel.h [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  // ----------------------------------------------------------------------
+
+  class ELseverityLevel {
+  public:
+    // ---  One ELseverityLevel is globally instantiated (see below)
+    // ---  for each of the following levels:
+    //
+    enum ELsev_ {
+      ELsev_noValueAssigned = 0  // default returned by map when not found
+      ,
```
Comment (makortel) [inline DataFormats/Common/interface/ELseverityLevel.h:33]: I understand this is a verbatim copy, but would it be worth to reformat
```suggestion
      ELsev_noValueAssigned = 0,  // default returned by map when not found
```
?

File: FWCore/MessageLogger/interface/ELseverityLevel.h [hunk 2]
```diff
@@ -18,120 +18,120 @@
...
+    // ----------------------------------------------------------------------
+
+    class ELseverityLevel {
+    public:
+      // ---  One ELseverityLevel is globally instantiated (see below)
+      // ---  for each of the following levels:
+      //
+      enum ELsev_ {
+        ELsev_noValueAssigned = 0  // default returned by map when not found
+        ,
```
Comment (makortel) [inline FWCore/MessageLogger/interface/ELseverityLevel.h:33]: If the other `ELseverityLevel::ELsev_` will be reformatted, reformat this one as well
```suggestion
        ELsev_noValueAssigned = 0,  // default returned by map when not found
```

**Comment by Dr15Jones** [comment]
please test

File: DataFormats/Common/interface/ELseverityLevel.h (hunk 1 above)
Comment (Dr15Jones) [reply DataFormats/Common/interface/ELseverityLevel.h]: done

File: FWCore/MessageLogger/interface/ELseverityLevel.h (hunk 2 above)
Comment (Dr15Jones) [reply FWCore/MessageLogger/interface/ELseverityLevel.h]: done

**Comment by davidlange6** [comment]
Thanks @Dr15Jones - this does remove a bunch of compilation errors (but it seems the rootcling traceback follows this change to DataFormats/Common..)

**Comment by Dr15Jones** [comment]
The test failures are false positives. Moving the dictionaries from FWCore/MessageLogger to DataFormats/Common has resulted in the following printouts in the log files

> Warning in <TInterpreter::ReadRootmapFile>: class  edm::ELseverityLevel found in libFWCoreMessageLogger.so  is already in libDataFormatsCommon.so 
> Warning in <TInterpreter::ReadRootmapFile>: class  edm::ErrorSummaryEntry found in libFWCoreMessageLogger.so  is already in libDataFormatsCommon.so 

This is because ROOT is seeing the dictionaries in FWCore/MessageLogger from the release and doesn't realize they are gone from the work area.

The tests which are failing are doing a 'diff' on the log from the job and are complaining about the two extra lines. Those lines will go away when this pull request is integrated into the release.

**Comment by makortel** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by slava77** [comment]
why do we still need `edm::messagelogger::ELseverityLevel` and `edm::messagelogger::ErrorSummaryEntry`?
The near-identity is quite confusing.

**Comment by slava77** [comment]
> why do we still need `edm::messagelogger::ELseverityLevel` and `edm::messagelogger::ErrorSummaryEntry`?
> The near-identity in implementation is quite confusing.

[in case the question got lost over the weekend]

**Comment by Dr15Jones** [comment]
> why do we still need edm::messagelogger::ELseverityLevel and edm::messagelogger::ErrorSummaryEntry?
The near-identity in implementation is quite confusing.

I want to keep FWCore/MessageLogger to be not dependent upon DataFormats/Common since MessageLogger is a 'low level' framework component. So edm::messagelogger::* are the objects MessageLogger uses.  The edm::* ones are those which have dictionaries and can be stored.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/32706	2e73826 
- code changes are essentially technical; they are in line with the PR description and the follow up clarification in https://github.com/cms-sw/cmssw/pull/32706#issuecomment-767580369; the code duplication seems OK for this somewhat exceptional case
- jenkins tests pass and comparisons with the baseline show no relevant differences

**Comment by davidlange6** [comment]
hi @dr15jones - did you consider a new DataFormats package instead of DataFormats/Common that keeps only the specific things needed by FWCOre/MessageLogger? FWCore/MessageLogger already depends on DataFormats/Provenance so it would not be a big change from today's situation.

**Comment by Martin-Grunewald** [comment]
+1

**Comment by silviodonato** [comment]
unassign analysis

**Comment by silviodonato** [comment]
@cms-sw/pdmv-l2 do you have any comments?

**Comment by chayanit** [comment]
+1

**Comment by silviodonato** [comment]
merge

**Comment by santocch** [comment]
+1

**Pull request merged**